CMakeLists.txt

```diff
@@ -417,7 +417,7 @@ target_link_libraries(executorch_no_prim_ops PRIVATE program_schema)
 # Check if dl exists for this toolchain and only then link it.
 find_library(DL_LIBRARY_EXISTS NAMES dl)
 # Check if the library was found
-if(DL_LIBRARY_EXISTS)
+if(DL_LIBRARY_EXISTS AND NOT EXECUTORCH_BUILD_CADENCE)
```

Comment:
Can you please elaborate on why this change was needed?

Reply:
Would love this to not bound to any specific backend though. Maybe we can have an option `EXECUTORCH_USE_DL` and you guys can set it to False.

Reply:
I agree, that would be better.

Reply:
updated the PR as per this suggestion